src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,33 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_CREDS_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new InvalidDataException("Login cannot be null");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidDataException("User already exists.");
+        }
+        if (user.getLogin().length() < MIN_CREDS_LENGTH) {
+            throw new InvalidDataException("Login must be at least 6 characters long.");
+        }
+        if (user.getPassword() == null) {
+            throw new InvalidDataException("Password cannot be null");
```

Comment:
Similar to the login validation, consider checking for both null and empty strings when validating the password.